google-apis-core/lib/google/apis/core/storage_upload.rb

```diff
@@ -234,25 +232,25 @@ def cancel_resumable_upload(client)
           request_header = header.dup
           request_header[CONTENT_LENGTH_HEADER] = '0'
           # Initiating call
-          response = client.delete(@upload_url, header: request_header, follow_redirect: true)
+          response = client.delete(@upload_url, nil, request_header)
           handle_resumable_upload_http_response_codes(response)
 
-          if !@upload_incomplete && (400..499).include?(response.code.to_i)
+          if !@upload_incomplete && (400..499).include?(response.status.to_i)
             @close_io_on_finish = true
             true # method returns true if upload is successfully cancelled
           else
-            logger.debug { sprintf("Failed to cancel upload session. Response: #{response.code} - #{response.body}") }
```

Comment:
I think the status is already either `int` or `nil`? and in case of `nil` this will be an empty string which is somewhat suboptimal for logs